kymatio/scattering3d/backend/numpy_backend.py

```diff
@@ -80,98 +58,22 @@ def compute_integrals(input_array, integral_powers):
     return integrals
 
 
-def fft(x, direction='C2C', inverse=False):
-    """
-        Interface with torch FFT routines for 2D signals.
-
-        Example
-        -------
-        x = torch.randn(128, 32, 32, 2)
-        x_fft = fft(x, inverse=True)
-
-        Parameters
-        ----------
-        input : tensor
-            complex input for the FFT
-        direction : string
-            'C2R' for complex to real, 'C2C' for complex to complex
-        inverse : bool
-            True for computing the inverse FFT.
-            NB : if direction is equal to 'C2R', then an error is raised.
-
-    """
-    if direction == 'C2R':
-        if not inverse:
-            raise RuntimeError('C2R mode can only be done with an inverse FFT.')
-
-    if direction == 'C2R':
-        output = np.real(ifftn(x, axes=(-3, -2, -1)))
-    elif direction == 'C2C':
-        if inverse:
-            output = ifftn(x, axes=(-3, -2, -1))
-        else:
-            output = fftn(x, axes=(-3, -2, -1))
-
-    return output
-
-
-def cdgmm3d(A, B, inplace=False):
-    """Complex pointwise multiplication.
-
-        Complex pointwise multiplication between (batched) tensor A and tensor B.
-
-        Parameters
-        ----------
-        A : torch tensor
-            Complex torch tensor.
-        B : torch tensor
-            Complex of the same size as A.
-        inplace : boolean, optional
-            If set True, all the operations are performed inplace.
-
-        Raises
-        ------
-        RuntimeError
-            In the event that the tensors are not compatibile for multiplication
-            (i.e. the final four dimensions of A do not match with the dimensions
-            of B), or in the event that B is not complex, or in the event that the
-            type of A and B are not the same.
-        TypeError
-            In the event that x is not complex i.e. does not have a final dimension
-            of 2, or in the event that both tensors are not on the same device.
-
-        Returns
-        -------
-        output : torch tensor
-            Torch tensor of the same size as A containing the result of the
-            elementwise complex multiplication of A with B.
-    """
-
-    if A.shape[-3:] != B.shape[-3:]:
-        raise RuntimeError('The tensors are not compatible for multiplication.')
-
-    if not _iscomplex(A) or not _iscomplex(B):
-        raise TypeError('The input, filter and output should be complex.')
-
-    if B.ndim != 3:
-        raise RuntimeError('The second tensor must be simply a complex array.')
-
-    if type(A) is not type(B):
-        raise RuntimeError('A and B should be same type.')
-
-
-    if inplace:
-        return np.multiply(A, B, out=A)
-    else:
-        return A * B
-
-
 def concatenate(arrays, L):
     S = np.stack(arrays, axis=1)
     S = S.reshape((S.shape[0], S.shape[1] // (L + 1), (L + 1)) + S.shape[2:])
     return S
 
 
+def rfft(x):
+    real_check(x)
+    return scipy.fftpack.fftn(x)
```

Comment:
how many axes does this choose by default?

Reply:
Appears to be all of them.

Reply:
That's not good…

Reply:
Does this get overridden by dimension-specific transforms maybe? And is actually completely unused?

Reply:
I specify the final three dims now.